File: grooves-api/src/main/groovy/com/github/rahulsom/grooves/api/AggregateType.java
package com.github.rahulsom.grooves.api;

/**
 * Marks a class as an aggregate.
 *
 * @param <AggregateIdT> The type of the primary key/identifier for the Aggregate
 *
 * @author Rahul Somasunderam
 */
public interface AggregateType<AggregateIdT> {
    AggregateIdT getId();
}

File: grooves-types/src/main/kotlin/com/github/rahulsom/grooves/api/AggregateType.kt
package com.github.rahulsom.grooves.api

/**
 * Marks a class as an aggregate.
 *
 * @param [AggregateIdT] The type of [AggregateType.id]
 *
 * @author Rahul Somasunderam
 */
interface AggregateType<out AggregateIdT> {
    /**
     * A unique identifier for the Aggregate.
     * This typically maps to the primary key in a persistence mechanism.
     */
    val id: AggregateIdT?
}
